src/App.js

```diff
@@ -33,21 +34,27 @@ function App() {
         <Route
           exact
           path="/teams"
+          user={currentUser}
           render={(routeProps) => <Teams {...routeProps} user={currentUser} />}
         />
-        <Route
-          exact
-          path="/teams/new"
-          render={(routeProps) => <AddTeam {...routeProps} user={currentUser} />}
-        />
+        <ProtectedRoute currentUser={currentUser} exact path="/teams/new">
+          <Route
+            exact
+            path="/teams/new"
+            render={(routeProps) => <AddTeam {...routeProps} user={currentUser} />}
+          />
+        </ProtectedRoute>
         <Route
           exact
           path="/teams/:id"
+          currentUser={currentUser}
           render={(routeProps) => <Team {...routeProps} user={currentUser} />}
         />
-        <Route exact path="/teams/:id/edit">
-          <EditTeam user={currentUser} />
-        </Route>
+        <ProtectedRoute currentUser={currentUser} exact path="/teams/:id/edit">
```

Comment:
This also shouldn't need the Route. Just the EditTeam component. Inside EditTeam it's grabbing what we would've sent with routerprops with a useParams() hook.